napari/utils/colormaps/colormap_utils.py

```diff
@@ -906,3 +906,52 @@ def display_name_to_name(display_name):
     return display_name_map.get(
         display_name, next(iter(AVAILABLE_COLORMAPS.keys()))
     )
+
+
+class CoercedContrastLimits(NamedTuple):
+    contrast_limits: Tuple[float, float]
+    offset: float
+    scale: float
+
+
+def _coerce_contrast_limits(contrast_limits: Tuple[float, float]):
+    """Coerce contrast limits to be in the float32 range."""
+    vispy_max = float(np.finfo(np.float32).max / 8)
+    if np.abs(contrast_limits).max() > vispy_max:
+        return scale_down(contrast_limits)
+
+    c_min = np.float32(contrast_limits[0])
+    if (
+        np.float32(contrast_limits[1]) - c_min
+        < (np.nextafter(c_min, np.float32(np.inf)) - c_min) * 64
+    ):
+        return scale_up(contrast_limits)
+
+    return CoercedContrastLimits(contrast_limits, 0, 1)
+
+
+def scale_down(contrast_limits: Tuple[float, float]):
+    """Scale down contrast limits to be in the float32 range."""
+    vispy_max = float(np.finfo(np.float32).max / 8)
+    # Using 8 as divisor comes from experiments.
```

Comment:
```suggestion
    # Using 8 as divisor comes from experiments.
```
Maybe put this comment above, where the magic 8 is first used.